Library/Core/Uno.Threading/Implementation/Posix/posix_thread.cpp

```diff
@@ -34,23 +26,13 @@ static void* ThreadStartup(void* arg)
     uDelegate* threadStart = @{Uno.Threading.Thread:Of(thread)._threadStart:Get()};
     assert(threadStart != NULL);
 
-#if IOS
     EnsureInitialized();
     pthread_setspecific(currentThread, (void*)thread);
```

Comment:
I believe we had bad crashes with this on old iOS devices. Something about pthread_setspecific not being supported on all iOS versions. It's only 1.5 years since that fix was introduced, so I'm a bit skeptical to changing it just like that...

This is 72ad247c07c918e23119ba8cf58638912859a3db in the old commit history (if you have access to that). I couldn't find a PR nor issue, but that might just be me being bad at searching.

Reply:
The code that was previously special-cased for iOS is now what we use on all Unix-platforms, a general implementation using pthread specifics instead of the `__thread` storage specifier for TLS.

I believe it must be the `__thread`-based code that caused those crashes, since we were using that on non-iOS Unix-platforms.

So this PR won't reintroduce bad crashes on iOS since the code is still the same.

Reply:
We're now using pthread specifics in the new cross-platform C++ wrapper: https://github.com/mortend/uno/blob/274644401421d7e35d07c0d127f2c0822179d2fc/Library/Core/UnoCore/Backends/CPlusPlus/Uno/Support.cpp#L128

The cross-platform wrapper is used by Uno in the static constructor here (previously `EnsureInitialized()`): https://github.com/mortend/uno/blob/274644401421d7e35d07c0d127f2c0822179d2fc/Library/Core/UnoCore/Source/Uno/Threading/Thread.uno#L44

We need the cross-platform wrapper since we also use it in `Memory.cpp` to handle auto-release pools etc.